featuretests/cmd_juju_status_test.go

```diff
@@ -92,10 +92,147 @@ func (s *StatusSuite) TestMultipleRelationsInYamlFormat(c *gc.C) {
+			Name:     "mysql",
+			Revision: "5",
+		}),
+	})
+	s.Factory.MakeUnit(c, &factory.UnitParams{
+		Application: application,
+		Machine:     machine,
+	})
+
+	context := s.run(c, "status")
+	c.Assert(cmdtesting.Stdout(context), jc.Contains, `
+App        Version  Status   Scale  Charm      Store       Rev  OS      Notes
+another             waiting    0/1  mysql      jujucharms    5  ubuntu  
+mysql               waiting    0/1  mysql      jujucharms    1  ubuntu  
+wordpress           waiting    0/1  wordpress  jujucharms    3  ubuntu  
+
+Unit         Workload  Agent       Machine  Public address  Ports  Message
+another/0    waiting   allocating  1                               waiting for machine
+mysql/0      waiting   allocating  0                               waiting for machine
+wordpress/0  waiting   allocating  0                               waiting for machine
+
+Machine  State    DNS  Inst id  Series   AZ  Message
+0        pending       id0      quantal      
+1        pending       id1      quantal      
+`)
+
+	context = s.run(c, "status", "0")
+	c.Assert(cmdtesting.Stdout(context), jc.Contains, `
+App        Version  Status   Scale  Charm      Store       Rev  OS      Notes
+mysql               waiting    0/1  mysql      jujucharms    1  ubuntu  
+wordpress           waiting    0/1  wordpress  jujucharms    3  ubuntu  
+
+Unit         Workload  Agent       Machine  Public address  Ports  Message
+mysql/0      waiting   allocating  0                               waiting for machine
+wordpress/0  waiting   allocating  0                               waiting for machine
+
+Machine  State    DNS  Inst id  Series   AZ  Message
+0        pending       id0      quantal      
+`)
+}
+
+func (s *StatusSuite) TestStatusFilteringByMachineIDMatchesExactly(c *gc.C) {
+	s.assertServeralUnitsOnAMachines(c)
+
+	application := s.Factory.MakeApplication(c, &factory.ApplicationParams{
+		Name:  "another",
+		Charm: s.Factory.MakeCharm(c, &factory.CharmParams{Name: "mysql"}),
+	})
+
+	// Put a unit from an application on the 1st machine.
+	machine1 := s.Factory.MakeMachine(c, &factory.MachineParams{
+		Jobs:       []state.MachineJob{state.JobHostUnits},
+		InstanceId: instance.Id("id1"),
+	})
+	s.Factory.MakeUnit(c, &factory.UnitParams{
+		Application: application,
+		Machine:     machine1,
+	})
+
+	// Burn machine numbers until we reach 10.
+
+	// Since machine 0 and 1 were created beforehand, we just need to
+	// create 7 machines here.
+	for i := 0; i < 8; i++ {
+		s.Factory.MakeMachine(c, &factory.MachineParams{
+			Jobs: []state.MachineJob{state.JobHostUnits},
+		})
+	}
+
+	// Put a unit from an application on the 10th machine.
+	machine10 := s.Factory.MakeMachine(c, &factory.MachineParams{
+		Jobs:       []state.MachineJob{state.JobHostUnits},
+		InstanceId: instance.Id("id10"),
+	})
+
+	s.Factory.MakeUnit(c, &factory.UnitParams{
+		Application: application,
+		Machine:     machine10,
+	})
+	context := s.run(c, "status", "1")
```

Comment:
Just to ensure and check there really is a machine 10, can we also run an check that "status 10" works as expected. Otherwise the test could pass by mistake.